src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm_the_fleshcrafter.cpp

```diff
@@ -55,19 +55,19 @@ class boss_salramm : public CreatureScript
 
         struct boss_salrammAI : public BossAI
         {
-            boss_salrammAI(Creature* creature) : BossAI(creature, DATA_SALRAMM)
-            {
-                Talk(SAY_SPAWN);
-            }
+            boss_salrammAI(Creature* creature) : BossAI(creature, DATA_SALRAMM) { }
+
+            void InitializeAI() override { Talk(SAY_SPAWN); }
 
             void EnterCombat(Unit* /*who*/) override
             {
                 Talk(SAY_AGGRO);
                 _EnterCombat();
 
-                events.ScheduleEvent(EVENT_CURSE_FLESH, 30000);
-                events.ScheduleEvent(EVENT_SUMMON_GHOULS, urand(19000, 24000));
-                events.ScheduleEvent(EVENT_SHADOW_BOLT, urand(8000, 12000));
+                events.ScheduleEvent(EVENT_CURSE_FLESH, Seconds(30));
+                events.ScheduleEvent(EVENT_SUMMON_GHOULS, randtime(Seconds(19),Seconds(24)));
+                events.ScheduleEvent(EVENT_SHADOW_BOLT, randtime(Seconds(8),Seconds(12)));
+                // heh
```

Comment:
the todo in original script is amusing

haven't gotten around to doing timers on boss scripts yet